s3.py

```diff
@@ -93,9 +92,29 @@ class S3Uploader(MirrorUploader):
         URL_TEMPLATE_DEFAULT: S3_BASE + u'%(bucket)s/%(key)s',
     }
 
+    PURPOSE = 'Purpose'
+    COVERS = 'Covers'
+    BOOKS = 'Books'
+    MARC = 'MARC'
+
     SETTINGS = [
         { "key": ExternalIntegration.USERNAME, "label": _("Access Key"), "required": True },
         { "key": ExternalIntegration.PASSWORD, "label": _("Secret Key"), "required": True },
+        { "key": PURPOSE, "label": _("Purpose"), "required": True,
```

Comment:
If I understand this correctly, under "Storage" a purpose for that storage integration is assigned.
Then, in the External Catalogs, you'd only see storages that should be used for MARC files. Similar for collections for covers and books storages.

The externalintegrationlink table would have something where the MARC exporter integration has a storage integration as its linked integration and the purpose is "MARC"?